tests/test_basic_s3.py

```diff
@@ -405,13 +441,25 @@ async def test_unicode_system_character(s3_client, bucket_name, create_object):
 
 @pytest.mark.moto
 @pytest.mark.asyncio
-async def test_non_normalized_key_paths(s3_client, bucket_name, create_object):
+async def test_non_normalized_key_paths(
+    s3_client,
+    bucket_name,
+    create_object,
+    current_http_backend: Literal['httpx', 'aiohttp'],
+):
     # The create_object method has assertEqual checks for 200 status.
     await create_object('key./././name')
     bucket = await s3_client.list_objects(Bucket=bucket_name)
     bucket_contents = bucket['Contents']
     assert len(bucket_contents) == 1
-    assert bucket_contents[0]['Key'] == 'key./././name'
+
+    # TODO: I don't know where the key normalization happens, if it's a problem that
+    # httpx doesn't normalize, or how to fix it if so.
+    key = bucket_contents[0]['Key']
+    if current_http_backend == 'httpx':
+        assert key == 'key./name'
+    else:
+        assert key == 'key./././name'
```

Comment:
hmm, ya this needs to be fixed.  We can't have the response changing.  This should match whatever botocore does, if current way is incorrect this is fine, otherwise needs to be fixed

Reply:
The current behaviour is indeed how botocore handles it:
https://github.com/boto/botocore/blob/970d577087d404b0927cc27dc57178e01a3371cd/tests/integration/test_s3.py#L599-L606
so I'll have to do some digging

Reply:
digging success!... but looks like it needs changes in upstream httpx to get fixed.

When we call `self._session.build_request` in
https://github.com/aio-libs/aiobotocore/blob/b19bc091cdca6eb019de9168b90f7c140b711102/aiobotocore/httpsession.py#L451-L456

httpx turns the string url into a `httpx.URL` object, which explicitly normalizes the path https://github.com/encode/httpx/blob/392dbe45f086d0877bd288c5d68abf860653b680/httpx/_urlparse.py#L387

We can manually create a `httpx.URL` object to be passed into `httpx.build_request`, but there's no parameter that control normalization, so at that point we'd have to create a subclass of `httpx.URL` or something to customize the behaviour.

I can open an issue for it in the httpx repo, but I'll first try to figure out why botocore seems to explicitly not want to normalize the path.

Reply:
the current percent encoding happens deep within the guts of botocore, where they explicitly mark `/` as safe for some reason: https://github.com/boto/botocore/blob/970d577087d404b0927cc27dc57178e01a3371cd/botocore/serialize.py#L520
Why? no clue. Can it be marked unsafe? I tried and ran unit tests and some functional tests in botocore/, and some tests did start to fail - though unclear if they're bad errors or just defensive asserts.

Reply:
This has been requested in httpx since forever: https://github.com/encode/httpx/discussions/1805
let's see if me offering to write a PR can give it some traction.

Reply:
ya main thing is downstream consumers may be doing string matching and could break logic

Reply:
appreciate the digging!

Reply:
turns out httpx does have a way to do it now! https://github.com/encode/httpx/discussions/1805#discussioncomment-8975989